packages/wdio-browserstack-service/src/reporter.ts

```diff
@@ -0,0 +1,80 @@
+import WDIOReporter from '@wdio/reporter'
+import { SuiteStats, TestStats, RunnerStats } from '@wdio/reporter'
+
+import { v4 as uuidv4 } from 'uuid'
+import { Options, Reporters } from '@wdio/types'
+import { BrowserstackConfig } from './types'
+import { getCloudProvider, uploadEventData, scopes } from './util'
+
+export default class TestReporter extends WDIOReporter {
+    private _capabilities: any
+    private _config: BrowserstackConfig & Options.Testrunner
+    private _observability?: boolean = true
+    private _sessionId?: string
+    private _suiteName?: string
+
+    constructor(options: Reporters.Options) {
+        super(options)
+        this._capabilities = {}
+        this._config = {} as any
+    }
+
+    onRunnerStart (runnerStats: RunnerStats) {
+        this._capabilities = runnerStats.capabilities
+        this._config = runnerStats.config
+        this._sessionId = runnerStats.sessionId
+        if (this._config.testObservability == false) this._observability = false
+    }
+
+    onSuiteStart (suiteStats: SuiteStats) {
+        this._suiteName = suiteStats.file
+    }
+
+    async onTestSkip (testStats: TestStats) {
```

Comment:
All our needs were getting done using service itself. The only issue was beforeTest and afterTest hooks don't run for tests which are marked as skipped (using `it.skip` for example) and just onTestSkip is called for them. So that's why only onTestSkip is needed here.